src/ConfigSpace/hyperparameters/distributions.py

```diff
@@ -0,0 +1,1301 @@
+    # std of a normal, centered on `center`, slowly expanding `std` such that
+    # rejection won't fail.
+
+    # We set up a buffer that can hold the number of neighbors we need, plus some
+    # extra excess from sampling, preventing us from having to reallocate memory.
+    # We also include the initial value in the buffer, as we will remove it later.
+    # OPTIM: If you really need to optimize this more, one could consider using `bins`
+    # as the more bins there are, the less need there is over sample for uniques
+    SAMPLE_SIZE = n * sample_multiplier
+    BUFFER_SIZE = n * (sample_multiplier + 1)
+    neighbors: Array[f64] = np.empty(BUFFER_SIZE + 1, dtype=f64)
+    neighbors[0] = qvector
+    offset = 1  # Indexes into current progress of filling buffer
+
+    # We extend the range of stds to try to find neighbors
+    if (
+        std == DEFAULT_VECTORIZED_NUMERIC_STD
+        and n_retries == NEIGHBOR_GENERATOR_N_RETRIES
+    ):
+        stds = _CACHED_LINSPACE
+    else:
+        stds = np.linspace(std, 1.0, n_retries + 1, endpoint=True)
+
+    range_size = upper - lower
+    for _std in stds:
+        # Generate candidates in vectorized space
+        candidates = seed.normal(qvector, _std * range_size, size=SAMPLE_SIZE)
+        valid_candidates = candidates[(candidates >= lower) & (candidates <= upper)]
+
+        # Transform to quantized space
+        candidates_quantized = quantize(
+            valid_candidates,
+            bounds=(lower, upper),
+            bins=bins,
+        )
+
+        uniq = np.unique(candidates_quantized)
+        new_uniq = np.setdiff1d(uniq, neighbors[:offset], assume_unique=True)
+
+        n_new_unique = len(new_uniq)
+        neighbors[offset : offset + n_new_unique] = new_uniq
+        offset += n_new_unique
+
+        # We have enough neighbors, we can stop
+        if offset - 1 >= n:
+            # Ensure we don't include the initial value point
+            return neighbors[1 : n + 1]
+
+    raise ValueError(
+        f"Failed to find enough neighbors with {n_retries} retries."
+        f" Given {n=} neighbors to generate, we only found {offset - 1}."
+        f" The normal's for sampling neighbors were Normal({vector}, {list(stds)})"
+        f" which were meant to find neighbors of {vector}. in the range"
+        f" ({lower}, {upper}).",
+    )
+
+
+def continuous_neighborhood(
+    vector: f64,
+    n: int,
+    *,
+    lower: f64,
+    upper: f64,
+    std: float | None = None,
+    seed: RandomState | None = None,
+    n_retries: int = NEIGHBOR_GENERATOR_N_RETRIES,
+    sample_multiplier: int = NEIGHBOR_GENERATOR_SAMPLE_MULTIPLIER,
+) -> Array[f64]:
+    """Create a neighborhood of `n` neighbors around `vector` with a normal distribution.
+
+    The neighborhood is created by sampling from a normal distribution centered around
+    `vector` with a standard deviation of `std`. The samples are then quantized to the
```

Comment:
Do you rather mean truncated?

Reply:
Also, there are no bins here, right?

Reply:
Yup, good spot, cleaned up the documentation for this.